contracts/token/CRC1155/presets/CRC1155PresetAutoId.sol

```diff
@@ -0,0 +1,158 @@
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mint(address to, uint256 amount, bytes memory data) public virtual {
+        mint(to, _tokenIdTracker.current(), amount, data);
+        _tokenIdTracker.increment();
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param count number of token IDs to mint, each token id has amount of 1.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256 count) public virtual {
+        uint256[] memory amounts = new uint256[](count);
+
+        for (uint i = 0; i < count; i++) {
+            amounts[i] = 1;
+        }
+
+        mintBatch(to, amounts);
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param amounts amount for each token ID to mint.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256[] memory amounts) public virtual {
+        mintBatch(to, amounts, "");
+    }
+
+    /**
+     * @dev Creates tokens with auto-generated token IDs in batch.
+     * @param to address to mint tokens.
+     * @param amounts amount for each token ID to mint.
+     * @param data Additional data used when `to` address is a contract.
+     * 
+     * Requirements:
+     *
+     * - the caller must have the `MINTER_ROLE`.
+     */
+    function mintBatch(address to, uint256[] memory amounts, bytes memory data) public virtual {
+        uint256[] memory tokenIds = new uint256[](amounts.length);
+
+        for (uint i = 0; i < amounts.length; i++) {
+            tokenIds[i] = _tokenIdTracker.current();
+            _tokenIdTracker.increment();
+        }
+
+        mintBatch(to, tokenIds, amounts, data);
+    }
+
+    /**
+     * @dev See {ERC1155-_beforeTokenTransfer}.
+     *
+     * Requirements:
+     *
+     * - each value in `amounts` should not be zero.
+     */
+    function _beforeTokenTransfer(
+        address operator,
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    )
+        internal virtual override(ERC1155PresetMinterPauser, CRC1155Enumerable)
+    {
+        for (uint i = 0; i < amounts.length; i++) {
+            require(amounts[i] > 0, "CRC1155Preset: amount is zero");
+        }
+
+        super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC1155PresetMinterPauser, CRC1155Enumerable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
```

Comment:
Personally I like this style more but I think common linters like solhint and solium will emit a warning that this line is too long. (Not sure if this happens if this contract is an npm dependency.)

Reply:
Currently, no lint tool configured. I suggest to auto-format the code at a time once lint tool configured.

On the other hand, the OZ [code](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/23869e5b2a7c6b9c3e27dee4289615b8cf50e36b/contracts/token/ERC1155/presets/ERC1155PresetMinterPauser.sol#L95) is also very long.

Reply:
Can this method be omitted? It just called the parent's method

Reply:
> On the other hand, the OZ code is also very long.

They changed this in their latest version for the same reason ([here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/eea376911b32d2c6392496d966e38601c1e762d5/contracts/token/ERC1155/presets/ERC1155PresetMinterPauser.sol#L105)). This can be addressed later once we have linting configured for this repo.

> Can this method be omitted? It just called the parent's method

I don't think it can, because it overrides the same method for two parent contracts (`ERC1155PresetMinterPauser` and `CRC1155Enumerable`).